ambry-store/src/main/java/com.github.ambry.store/BlobStore.java

```diff
@@ -612,6 +615,70 @@ public void updateTtl(MessageWriteSet messageSetToUpdate) throws StoreException
     }
   }
 
+  @Override
+  public short undelete(MessageInfo info) throws StoreException {
+    checkStarted();
+    final Timer.Context context = metrics.undeleteResponse.time();
+    try {
+      StoreKey id = info.getStoreKey();
+      Offset indexEndOffsetBeforeCheck = index.getCurrentEndOffset();
+      List<IndexValue> values = index.findAllIndexValuesForKey(id, null);
+      index.validateSanityForUndelete(id, values, IndexValue.LIFE_VERSION_FROM_FRONTEND);
+      IndexValue latestValue = values.get(0);
```

Comment:
It's clear when oldest/latest is used in variable. I would like to suggest also use var oldest/latest in validateSanityForUndelete, instead of first/last. To be consistent.